synapse/appservice/__init__.py

```diff
@@ -94,7 +94,9 @@ def __init__(
         ip_range_whitelist=None,
     ):
         self.token = token
-        self.url = url
+        self.url = (
+            None if url is None else url.rstrip("/")
```

Comment:
I might change this to:
```suggestion
            url.rstrip("/") if isinstance(url, str) else None
```

And add a comment for why this is necessary.

Reply:
Isn't there one below?